src/UseCases/PullChangesFromCyclopsUseCase.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Gcd\Cyclops\UseCases;
+
+use Gcd\Cyclops\Exceptions\CyclopsException;
+use Gcd\Cyclops\Services\CyclopsService;
+
+class PullChangesFromCyclopsUseCase
+{
+    /**
+     * @var CyclopsService
+     */
+    private $cyclopsService;
+
+    public function __construct(CyclopsService $cyclopsService)
+    {
+        $this->cyclopsService = $cyclopsService;
+    }
+
+    public function execute(string $changesSince, callable $setOptIn)
+    {
+        try {
+            $statusChanges = $this->cyclopsService->getBrandOptInStatusChanges($changesSince);
+
+            foreach ($statusChanges as $cyclopsId => $optIn) {
+                $setOptIn($cyclopsId, $optIn);
+            }
+
+            if (file_exists('/cyclops/last-status-changes-date.txt')) {
```

Comment:
i wouldn't do any of this in the use case. This is all environment specific storage and retrieval, I'd suggest that the command could do this (but not in the root directory, work relative to the working directory). In my opinion the use cae should just get in a `\DateTime` object and return a `\DateTime`, which the command can take care of storing and providing each time.